cloud/common/deploy/provider/terraform.go

```diff
@@ -241,32 +267,39 @@ func createTerraformStackForNitricProvider(req *deploymentspb.DeploymentUpReques
 	if ok && len(backend) == 1 {
 		backendType := lo.Keys(backend)[0]
 		config := backend[backendType].(map[string]interface{})
-		jsonMap, err := json.Marshal(config)
 		if err != nil {
 			logger.Fatalf("Failed to serialize backend config %v", err)
 			return err
 		}
 
 		switch backendType {
 		case "gcs":
-			gcsConfig := &cdktf.GcsBackendConfig{}
-			// serialize the backend config
-			err := json.Unmarshal(jsonMap, gcsConfig)
-			if err != nil {
-				return err
-			}
-			cdktf.NewGcsBackend(stack, gcsConfig)
+			err = configureTfBackend(stack, config, cdktf.NewGcsBackend)
 		case "s3":
-			s3Config := &cdktf.S3BackendConfig{}
-			// serialize the backend config
-			err := json.Unmarshal(jsonMap, s3Config)
-			if err != nil {
-				return err
-			}
-			cdktf.NewS3Backend(stack, s3Config)
+			err = configureTfBackend(stack, config, cdktf.NewS3Backend)
+		case "azurerm":
+			err = configureTfBackend(stack, config, cdktf.NewAzurermBackend)
+		case "http":
+			err = configureTfBackend(stack, config, cdktf.NewHttpBackend)
+		case "remote":
+			err = configureTfBackend(stack, config, cdktf.NewRemoteBackend)
+		case "local":
+			err = configureTfBackend(stack, config, cdktf.NewLocalBackend)
+		case "consul":
+			err = configureTfBackend(stack, config, cdktf.NewConsulBackend)
+		case "cos":
+			err = configureTfBackend(stack, config, cdktf.NewCosBackend)
+		case "oss":
+			err = configureTfBackend(stack, config, cdktf.NewOssBackend)
+		case "pg":
+			err = configureTfBackend(stack, config, cdktf.NewPgBackend)
```

Comment:
tried including these in a generic map, but it was a non-starter so have just made it a switch instead.